install/terraform/modules/aks/aks.tf

```diff
@@ -127,6 +127,20 @@ resource "azurerm_network_security_rule" "gameserver" {
   network_security_group_name = azurerm_network_security_group.agones_sg.name
 }
 
+resource "azurerm_network_security_rule" "gameserver" {
```

Comment:
Can we remove this for now?

Two reasons:
1) It's a bit outside the scope of this PR I think. Likely better handled in it's own PR.
2) We likely don't want to default to having TCP open in firewall rules. We should probably look at how we can make firewall config through terraform configurable (and hence, it's own PR)

There's probably a longer discussion that needs to be had around Terrform config and what should be default and what shouldn't be - and I don't want to see it hold up your work here. (This is also why smaller PRs are better in general).

If we want to change things for the e2e cluster, let's add it [here](https://github.com/googleforgames/agones/blob/master/build/terraform/e2e/module.tf), with an extra `gke-tcp-game-server-firewall` firewall configuration. This is the script that is run to create our e2e cluster and it's configurations.

Reply:
Makes perfect sense - I will remove the terraform changes and add it in the e2e config.

Reply:
But should the TCP rule not be allowed with this? Or does it work differently on the cloud build, which would cause the test to fail?